### PR TITLE
Package imagestream

### PR DESCRIPTION
**Pull request by dmage**

### CODE REVIEW

**Comment by bparees** [comment]
unless this is needed for something urgent, please hold it until i land the origin dep removal.

**Comment by dmage** [comment]
Sure, I’m going to make it as a set of small commits to simplify rebase on your changes.

File: pkg/dockerregistry/server/repository.go [hunk 1]
```diff
@@ -79,22 +75,29 @@ func (app *App) Repository(ctx context.Context, repo distribution.Repository, cr
...
+			imageStreamGetter: &cachedImageStreamGetter{
+				ctx:          ctx,
+				namespace:    namespace,
+				name:         name,
+				isNamespacer: registryOSClient,
 			},
 		},
 	}
 
+	r.cache = &cache.RepoDigest{
```
Comment (legionus) [inline pkg/dockerregistry/server/repository.go]: Why is it moved here ?
```go
r := &repository{
   ...
   cache: &cache.RepoDigest{
      Cache: app.cache,
   },
}
```
???

File: pkg/dockerregistry/server/imagestream/imagestream.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+type ImageStream interface {
+	Reference() string
+	Exists() (bool, error)
+
+	GetImageOfImageStream(ctx context.Context, dgst digest.Digest) (*imageapiv1.Image, *imageapiv1.ImageStream, error)
+	CreateImageStreamMapping(ctx context.Context, tag string, image *imageapiv1.Image) error
+	ImageManifestBlobStored(ctx context.Context, image *imageapiv1.Image) error
+
+	HasBlob(ctx context.Context, dgst digest.Digest, requireManaged bool) *imageapiv1.Image
+	IdentifyCandidateRepositories(primary bool) ([]string, map[string]ImagePullthroughSpec, error)
```
Comment (legionus) [inline pkg/dockerregistry/server/imagestream/imagestream.go]: `IdentifyCandidateRepositories` -> `CandidateRepositories` ?

File: pkg/dockerregistry/server/repository.go (hunk 1 above)
Comment (dmage) [reply pkg/dockerregistry/server/repository.go]: fixed

**Pull request merged**